src/util.js

```diff
@@ -81,44 +81,69 @@ export function phrase (me, id, ...args) {
 	return id + " " + args.join(" ");
 }
 
-/**
- * Test whether a URL matches a set of criteria.
- * @param {string | URL} url
- * @param {{protocol, host, urls} | {protocol, host, urls}[]} criteria
- * @returns {boolean}
- */
-export function testURL (url, criteria) {
-	if (typeof url === "string") {
-		url = new URL(url);
+export const URLPattern = globalThis.URLPattern ?? import("../lib/urlpattern-polyfill/index.js");
+
+export function testURLs (source, urls) {
+	if (!urls) {
+		return false;
 	}
 
-	if (Array.isArray(criteria)) {
-		return criteria.some(pattern => testURL(url, pattern));
+	for (let i = 0; i < urls.length; i++) {
+		let url = urls[i];
+
+		if (!(url instanceof URLPattern)) {
+			url = urls[i] = new URLPattern(url);
+		}
+
+		if (url.test(source)) {
+			return true;
+		}
 	}
 
-	let {protocol, host, path} = criteria;
+	return false;
+}
 
-	if (protocol && url.protocol !== protocol) {
-		return false;
+export function matchURLs (source, urls) {
+	if (!urls) {
+		return null;
 	}
 
-	if (host) {
-		if (host.startsWith("*.")) {
-			// Wildcard subdomain
-			host = host.replace(/^(\*\.)+/, "");
+	for (let i = 0; i < urls.length; i++) {
+		let url = urls[i];
 
-			if (url.host !== host && !url.host.endsWith("." + host)) {
-				return false;
-			}
+		if (!(url instanceof URLPattern)) {
+			url = urls[i] = new URLPattern(url);
 		}
-		else if (url.host !== host) {
-			return false;
+
+		let match = matchURL(source, url);
+
+		if (match) {
+			return match;
 		}
 	}
 
-	if (path && !url.pathname.startsWith(path)) {
-		return false;
+	return null;
+}
+
+export function matchURL (source, urlPattern) {
+	let match = urlPattern.exec(source);
```

Comment:
```suggestion
	source = source.replace(/\n\r?/g, " "); // Why? url.hash drops line breaks, which might be delimiters we should preserve

	let match = urlPattern.exec(source);
```

Reply:
If we have this, we can simplify the Github API's `parseURL()` when working with GraphQL.